models/user/external_login_user.go

```diff
@@ -160,12 +160,34 @@ func UpdateExternalUserByExternalID(ctx context.Context, external *ExternalLogin
 	return err
 }
 
+// EnsureLinkExternalToUser link the external user to the user
+func EnsureLinkExternalToUser(external *ExternalLoginUser) error {
```

Comment:
Instead of db.DefaultContext could you pass the ctx throug?

Reply:
Yes confirmed this could case a deadlock!

Reply:
Fixed in https://github.com/go-gitea/gitea/pull/31572/commits/2b71ec7a225b215632e498306060870de6295451